### ISSUE TITLE
Docker image fails to build

### ISSUE THREAD

**Issue by ChrisTimperley** [rosin-project/rosinstall_generator_time_machine#14]
I suspect that a transitive dependency no longer supports Python 2.

```
Step 15/19 : RUN /rgtm/pre141/bin/pip install -U pip  && /rgtm/pre141/bin/pip install -U wheel setuptools==20.1.1  && /rgtm/pre141/bin/pip install       git+https://github.com/rosin-project/rosdistro_python@rosin_bughunt_0.2.20#egg=rosdistro       rosinstall==0.7.2       rosdep==0.10.24       rosinstall_generator==0.1.5  && rm -rf /root/.cache/pip
 ---> Running in 8b25c17bc761
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Requirement already up-to-date: pip in ./pre141/lib/python2.7/site-packages (20.0.2)
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Requirement already up-to-date: wheel in ./pre141/lib/python2.7/site-packages (0.34.2)
Collecting setuptools==20.1.1
  Downloading setuptools-20.1.1-py2.py3-none-any.whl (472 kB)
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 45.0.0
    Uninstalling setuptools-45.0.0:
      Successfully uninstalled setuptools-45.0.0
Successfully installed setuptools-20.1.1
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting rosdistro
  Cloning https://github.com/rosin-project/rosdistro_python (to revision rosin_bughunt_0.2.20) to /tmp/pip-install-FOGdDl/rosdistro
  Running command git clone -q https://github.com/rosin-project/rosdistro_python /tmp/pip-install-FOGdDl/rosdistro
  Running command git checkout -b rosin_bughunt_0.2.20 --track origin/rosin_bughunt_0.2.20
  Switched to a new branch 'rosin_bughunt_0.2.20'
  Branch rosin_bughunt_0.2.20 set up to track remote branch rosin_bughunt_0.2.20 from origin.
Collecting rosinstall==0.7.2
  Downloading rosinstall-0.7.2.tar.gz (24 kB)
Collecting rosdep==0.10.24
  Downloading rosdep-0.10.24.tar.gz (67 kB)
Collecting rosinstall_generator==0.1.5
  Downloading rosinstall_generator-0.1.5.tar.gz (8.0 kB)
Requirement already satisfied: argparse in /usr/lib/python2.7 (from rosdistro) (1.2.1)
Collecting catkin_pkg
  Downloading catkin_pkg-0.4.16.tar.gz (60 kB)
Collecting distribute
  Downloading distribute-0.7.3.zip (145 kB)
Collecting rospkg
  Downloading rospkg-1.2.3.tar.gz (41 kB)
Collecting PyYAML
  Downloading PyYAML-5.3.tar.gz (268 kB)
Collecting vcstools>=0.1.30
  Downloading vcstools-0.1.42.tar.gz (54 kB)
    ERROR: Command errored out with exit status 1:
     command: /rgtm/pre141/bin/python2.7 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-FOGdDl/vcstools/setup.py'"'"'; __file__='"'"'/tmp/pip-install-FOGdDl/vcstools/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-FOGdDl/vcstools/pip-egg-info
         cwd: /tmp/pip-install-FOGdDl/vcstools/
    Complete output (1 lines):
    error in vcstools setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
The command '/bin/sh -c /rgtm/pre141/bin/pip install -U pip  && /rgtm/pre141/bin/pip install -U wheel setuptools==20.1.1  && /rgtm/pre141/bin/pip install       git+https://github.com/rosin-project/rosdistro_python@rosin_bughunt_0.2.20#egg=rosdistro       rosinstall==0.7.2       rosdep==0.10.24       rosinstall_generator==0.1.5  && rm -rf /root/.cache/pip' returned a non-zero code: 1
```

**Comment by ChrisTimperley** [comment]
This particular problem can be avoided by pinning the version of `vcstools` in the Dockerfile (or using a requirements.txt).
```
vcstools==0.1.30 
```

Once that change is made, the build proceeds to the next step but encounters a new failure:
```
 ---> 5669287f992b
Step 16/19 : RUN /rgtm/post141/bin/pip install -U pip  && /rgtm/post141/bin/pip install -U wheel setuptools  && /rgtm/post141/bin/pip install       git+https://github.com/rosin-project/rosdistro_python@rosin_bughunt_0.6.8#egg=rosdistro       rosinstall_generator  && rm -rf /root/.cache/pip
 ---> Running in b28e55b1131c
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Requirement already up-to-date: pip in ./post141/lib/python2.7/site-packages (20.0.2)
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Requirement already up-to-date: wheel in ./post141/lib/python2.7/site-packages (0.34.2)
Requirement already up-to-date: setuptools in ./post141/lib/python2.7/site-packages (45.0.0)
ERROR: Package 'setuptools' requires a different Python: 2.7.12 not in '>=3.5'
The command '/bin/sh -c /rgtm/post141/bin/pip install -U pip  && /rgtm/post141/bin/pip install -U wheel setuptools  && /rgtm/post141/bin/pip install       git+https://github.com/rosin-project/rosdistro_python@rosin_bughunt_0.6.8#egg=rosdistro       rosinstall_generator  && rm -rf /root/.cache/pip' returned a non-zero code: 1
```

**Comment by ChrisTimperley** [comment]
`setuptools 44.0.0` was the last version to support Python 2.7:
https://pypi.org/project/setuptools/44.0.0/

**Comment by gavanderhoorn** [comment]
Yes, I'm aware of this.

`setuptools` can be limited to a version which support Python 2.7 as well.

That should be the last difficult package though, IIRC.

**Comment by gavanderhoorn** [comment]
I have the image locally on my computer. I could export it for you and share it somewhere.

It's going to take a bit of time though: the image is about 330MB and I'm on my home connection right now.

**Comment by ChrisTimperley** [comment]
No worries. I've got the image to build on my machine.
Another possibility for the future is to setup an automated build on DockerHub, although that might be overkill.

**Comment by gavanderhoorn** [comment]
Seeing as we're enabling ppl to build historic images I doubt it'd be necessary to rebuild the image often. I'll see if I setup a `rosin-project` Dockerhub account and push the image I have to it.

**Comment by gavanderhoorn** [comment]
I believe this was fixed in #15, so closing.
